_std/defines/component_defines/component_defines_atom.dm

```diff
@@ -157,6 +157,31 @@
 	/// When removed
 	#define COMSIG_ITEM_IMPLANT_REMOVED "implant_removed"
 
+	// ---- wire panel signals ----
+
+	/// When a mob attempts an action on a wire (mob/user, wire, action)
+	#define COMSIG_WPANEL_MOB_WIRE_ACT "wpanel_mob_wire_act"
+	/// Ion storm handler ()
+	#define COMSIG_WPANEL_ION_STORM "wpanel_ion_storm"
+	/// Disable Random Wire ()
+	#define COMSIG_WPANEL_DISABLE_RANDOM_WIRE "wpanel_disable_random_wire"
+
+	/// Set the cover status `WPANEL_COVER_*` (mob/user, status)
+	#define COMSIG_WPANEL_SET_COVER "wpanel_set_cover"
+	/// Set a wire control to be TRUE or FALSE (mob/user, controls, new_status)
+	#define COMSIG_WPANEL_SET_CONTROL "wpanel_set_control"
+
+	/// TGUI helper; call in `ui_data` and pass in a list (mob/user, list/data)
+	#define COMSIG_WPANEL_UI_DATA "wpanel_ui_data"
+	/// TGUI helper; call in `ui_static_data` and pass in a list (mob/user, list/data)
+	#define COMSIG_WPANEL_UI_STATIC_DATA "wpanel_ui_static_data"
+	/// TGUI helper; call in `ui_act` (action, list/params, datum/tgui/ui)
+	#define COMSIG_WPANEL_UI_ACT "wpanel_ui_act"
+
+	/// Returns state `WPANEL_COVER_*` of cover ()
+	#define COMSIG_WPANEL_STATE_COVER "wpanel_state_cover"
+ 	/// Returns bitmask of active wire controls ()
+	#define COMSIG_WPANEL_STATE_CONTROLS "wpanel_state_controls"
```

Comment:
Wouldn't it make more sense for a lot of these to be generic TGUI signals? That would reduce the amount of boilerplate you'd have to write for each implementation. (and I don't think it would have a noticeable performance impact?)

Reply:
Can you help me understand more about making these more generic? Just merging controls and cover, or something else?

Reply:
Instead of `COMSIG_WPANEL_UI_DATA` couldn't it just be `COMSIG_ATOM_UI_DATA` sent in the `ui_data` parent or caller proc?

Reply:
Can you take a look at [28cfa3d](https://github.com/goonstation/goonstation/pull/13624/commits/28cfa3dc48be2b07243f6ec0e6baa81458f84d89) - I think it's what you had in mind but it feels like an invasive change

Reply:
@ZeWaka This seems like the cleanest solution to me, but I think you said something about it being a bad idea? Apologies if I'm pinging the wrong person.